examples/TransactionGenerator.hs

```diff
@@ -241,178 +256,188 @@ scriptConfigParser = id
-  , _genClientEnv       :: ClientEnv
-  , _genChainId         :: ChainId
-  , _genVersion         :: ChainwebVersion
+  { _confTimingDist :: !(Maybe TimingDistribution)
+  , _confKeysets    :: !(Map Account (Map ContractName [SomeKeyPair]))
+  , _confClientEnv  :: !ClientEnv
+  , _confVersion    :: !ChainwebVersion
   }
 
-makeLenses ''TXGConfig
+confKeysets :: Lens' TXGConfig (Map Account (Map ContractName [SomeKeyPair]))
+confKeysets f c = (\ks -> c { _confKeysets = ks }) <$> f (_confKeysets c)
 
-generateDelay :: (MonadIO m) => TXG m Int
+generateDelay :: MonadIO m => TXG m Int
 generateDelay = do
-  distribution <- view timingdist
-  gen <- use gsGen
+  distribution <- asks _confTimingDist
+  gen <- gets _gsGen
   case distribution of
     Just (Gaussian gmean gvar) -> liftIO (truncate <$> normal gmean gvar gen)
     Just (Uniform ulow uhigh)  -> liftIO (truncate <$> uniformR (ulow, uhigh) gen)
     Nothing                    -> error "generateDelay: impossible"
 
-generateSimpleTransaction :: (MonadIO m, MonadLog SomeLogMessage m) => TXG m (Command Text)
+generateSimpleTransaction
+  :: (MonadIO m, MonadLog SomeLogMessage m)
+  => TXG m (ChainId, Command Text)
 generateSimpleTransaction = do
   delay <- generateDelay
   stdgen <- liftIO newStdGen
   let (operandA, operandB, op) =
         flip evalState stdgen $ do
-            a <- state (randomR (1, 100 :: Integer))
-            b <- state (randomR (1, 100 :: Integer))
-            ind <- state (randomR (0, 2 :: Int))
+            a <- state $ randomR (1, 100 :: Integer)
+            b <- state $ randomR (1, 100 :: Integer)
+            ind <- state $ randomR (0, 2 :: Int)
             let operation = "+-*" !! ind
-            return (a, b, operation)
+            pure (a, b, operation)
       theCode = "(" ++ [op] ++ " " ++ show operandA ++ " " ++ show operandB ++ ")"
-  cid <- view genChainId
+
+  -- Choose a Chain to send this transaction to, and cycle the state.
+  cid <- uses gsChains NES.head
+  gsChains %= rotate
+
+  -- Delay, so as not to hammer the network.
   liftIO $ threadDelay delay
-  lift $ logg Info $ toLogMessage $ T.pack $ "The delay is" ++ show delay ++ " seconds."
-  lift $ logg Info $ toLogMessage $ T.pack $ "Sending expression " ++ theCode
+  -- lift . logg Info . toLogMessage . T.pack $ "The delay is " ++ show delay ++ " seconds."
+  lift . logg Info . toLogMessage . T.pack $ printf "Sending expression %s to %s" theCode (show cid)
   kps <- liftIO testSomeKeyPairs
+
+
   let publicmeta = CM.PublicMeta (CM.ChainId $ chainIdToText cid) "sender00" (ParsedInteger 100) (ParsedDecimal 0.0001)
       theData = object ["test-admin-keyset" .= fmap formatB16PubKey kps]
-  liftIO $ mkExec theCode theData publicmeta kps Nothing
+  cmd <- liftIO $ mkExec theCode theData publicmeta kps Nothing
+  pure (cid, cmd)
+
+-- | O(1). The head value is moved to the end.
+rotate :: NESeq a -> NESeq a
+rotate (h :<|| rest) = rest :||> h
 
+-- TODO What is this number?
```

Comment:
We might want to take this out so as to not confused anyone else who modifies this code in the future. The contracts we want to test are well established at this point.